### PR TITLE
Add support for url branding feature

### PR DESCRIPTION
**Pull request by sadilchamishka**
## Purpose
$subject

## Related Issues
- https://github.com/wso2/product-is/issues/14099

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManager.java [hunk 1]
```diff
@@ -43,11 +43,19 @@ void shareOrganizationApplication(String ownerOrgId, String mainAppId, List<Stri
...
-    ServiceProvider resolveSharedApplication(String mainAppName, String ownerOrgId, String sharedOrgId)
+    ServiceProvider resolveSharedApplication(String mainAppName, String sharedOrgId)
             throws OrganizationManagementException;
+
+    /**
+     * Check if SAAS app registered on given tenant
+     *
+     * @param mainAppName Name of the main application
+     * @param tenantDomain Tenant
+     * @return
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManager.java:58]: Describe the return of the method. and end the method description and param descriptions with full stop

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 2]
```diff
@@ -109,33 +107,33 @@ public void shareOrganizationApplication(String ownerOrgId, String originalAppId
...
-        String sharedOrgTenantDomain = getOrganizationManager().resolveTenantDomain(sharedOrgId);
         try {
-            return getApplicationManagementService().getApplicationByResourceId(sharedAppId, sharedOrgTenantDomain);
+            ServiceProvider mainApplication = getApplicationManagementService()
+                    .getServiceProvider(mainAppName, tenantDomain);
+            if (mainApplication != null) {
+                return true;
+            } else {
+                return false;
+            }
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:134]:
```suggestion
            if (mainApplication != null) {
                return true;
            }
            return false;
```

File: components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java [hunk 3]
```diff
@@ -231,29 +233,63 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
...
+            // #todo remove this workaround after find a way to capture orgId value "o/<org-id>"
+            String orgId = IdentityTenantUtil.getTenantDomainFromContext();
+            if (orgId == null || SUPER_TENANT_DOMAIN_NAME.equals(orgId)) {
+                orgId = getOrgIdByTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+            }
+
+            String tenantDomain = getTenantDomain(orgId);
+            // Check the request came for SAAS app owned org. If so redirect to org name capture.
+            if (isSpAppOwnedByTenant(context.getServiceProviderName(), tenantDomain)) {
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java:262]: remove additional new lines

File: components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java [hunk 4]
```diff
@@ -231,29 +233,63 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
...
+            runtimeParams.put(ORG_ID_PARAMETER, org);
+            runtimeParams.put(ORG_NAME_PARAMETER, orgName);
         } catch (OrganizationManagementException e) {
-            throw handleAuthFailures(ERROR_CODE_ERROR_CHECKING_ORGANIZATION_EXIST_BY_ID, e);
+            try {
+                // Also, org name can be provided. ex - From org name capture login page.
+                String orgId = getOrganizationManager().getOrganizationIdByName(org);
+                runtimeParams.put(ORG_NAME_PARAMETER, org);
+                runtimeParams.put(ORG_ID_PARAMETER, orgId);
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java:288]: remove additional new line

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManager.java (hunk 1 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManager.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/77/commits/bc2b289d2f943096afa0ea26e5fce05a70dec463

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java (hunk 2 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/77/commits/bc2b289d2f943096afa0ea26e5fce05a70dec463

File: components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java (hunk 3 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/77/commits/bc2b289d2f943096afa0ea26e5fce05a70dec463

File: components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java [hunk 5]
```diff
@@ -231,29 +233,63 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
...
+            String tenantDomain = getTenantDomain(orgId);
+            // Check the request came for SAAS app owned org. If so redirect to org name capture.
+            if (isSpAppOwnedByTenant(context.getServiceProviderName(), tenantDomain)) {
+
+                redirectToOrgNameCapture(response, context);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            } else {
+                resolveOrgNameAndId(orgId, context);
+            }
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/77/commits/bc2b289d2f943096afa0ea26e5fce05a70dec463

File: components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java (hunk 4 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.authn/src/main/java/org/wso2/carbon/identity/organization/management/application/authn/EnterpriseIDPAuthenticator.java]: Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/77/commits/bc2b289d2f943096afa0ea26e5fce05a70dec463

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java [hunk 6]
```diff
@@ -328,6 +339,18 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
...
 
+    @Override
+    public List<Organization> getOrganizationsByName(String organizationName)
+            throws OrganizationManagementException {
+
+        List<Organization> organizations = organizationManagementDAO.getOrganizationsByName(organizationName);
+        if (CollectionUtils.isNotEmpty(organizations)) {
+            return organizations;
+        } else {
+            throw handleClientException(ERROR_CODE_ERROR_RETRIEVING_ORGANIZATIONS_BY_NAME, organizationName);
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java:350]: define client error code for invalid org name

**Comment by sadilchamishka** [comment]
Fixed by - https://github.com/wso2-extensions/identity-organization-management/pull/103

**Comment by sadilchamishka** [comment]
Fixed by : https://github.com/wso2-extensions/identity-organization-management/pull/103

**Pull request closed without merging**